hearth/lib/hearth/middleware/send.rb

```diff
@@ -32,6 +32,9 @@ def call(input, context)
           stub = @stubs.next(context.operation_name)
           output = Output.new
           apply_stub(stub, input, context, output)
+          if context.response.body.respond_to?(:rewind)
```

Comment:
I think this should be the stubs job, to rewind itself if we are writing out a string io.

Reply:
Yeah - I'm not against pushing it to the stubs.  But it means we have to do it N places intstead of 1.  And things will be broken if we miss it in even one place.  

The second reason I opted for it here was to match the http client behavior (ie, its the responsibility of our http client to rewind the body, not the responsibility of the library that code depends on).